src/components/ProductItem/ProductItem.tsx

```diff
@@ -49,6 +54,16 @@ const ProductItem: React.FC<ProductItemProps> = ({
     handleDecrementItem(id);
   };
 
+  const handleAddToFavorite = () => {
+    const isAlreadyAdded = favoriteItems.some(
```

Comment:
We decided to use method from context, didn't we?

Reply:
fixed